js/piwik.js

```diff
@@ -3476,17 +3480,43 @@ if (typeof window.Piwik !== 'object') {
              * The infamous web bug (or beacon) is a transparent, single pixel (1x1) image
              */
             function getImage(request, callback) {
-                var image = new Image(1, 1);
+                // make sure to actually load an image so callback gets invoked
+                request = request.replace("send_image=0","send_image=1");
 
+                var image = new Image(1, 1);
                 image.onload = function () {
                     iterator = 0; // To avoid JSLint warning of empty block
                     if (typeof callback === 'function') { callback(); }
                 };
-                // make sure to actually load an image so callback gets invoked
-                request = request.replace("send_image=0","send_image=1");
                 image.src = configTrackerUrl + (configTrackerUrl.indexOf('?') < 0 ? '?' : '&') + request;
             }
 
+            function sendPostRequestViaSendBeacon(request)
+            {
+                var supportsSendBeacon = 'object' === typeof navigatorAlias
+                    && 'function' === typeof navigatorAlias.sendBeacon
+                    && 'function' === typeof Blob;
+
+                if (!supportsSendBeacon) {
+                    return false;
+                }
+
+                var headers = {type: 'application/x-www-form-urlencoded; charset=UTF-8'};
+                var success = false;
+
+                try {
+                    var blob = new Blob([request], headers);
+                    success = navigator.sendBeacon(configTrackerUrl, blob);
```

Comment:
When sending the beacon, currently  we don't get the data in the GET, but as POST. since it's useful to get GET requests so we can replay the logs, wondering if we should maybe do the same as in `getImage` and write eg. `navigator.sendBeacon( configTrackerUrl + (configTrackerUrl.indexOf('?') < 0 ? '?' : '&') + request, blob)`  and send the data as GET params instead of POST?

Reply:
@mattab  see https://github.com/matomo-org/matomo/issues/6641#issuecomment-364261369

It is not possible to send a GET request but mentioned in #6641 we need to check if log importer can process POST request when URL parameters are used